model/task/task.go

```diff
@@ -356,16 +356,38 @@ func (t *Task) SetOverrideDependencies(userID string) error {
 }
 
 func (t *Task) AddDependency(d Dependency) error {
+	query := bson.M{IdKey: t.Id}
+	update := bson.M{
+		"$push": bson.M{
+			DependsOnKey: d,
+		},
+	}
+	for _, existingDependency := range t.DependsOn {
+		if existingDependency.TaskId == d.TaskId && existingDependency.Status == d.Status {
+			// If the dependency in the DB is attainable and we want to set it unattainable, update dependency.
```

Comment:
if there's already a dependency on the earlier task in the taskgroup wouldn't it have been marked unattainable by `markEnd`?

Reply:
I'm not sure exactly. AddDependency is called from BlockTaskGroupTasks which is called in the APIServer's EndTask (where MarkEnd is called after) and the TaskExecutionTimeoutJob where MarkEnd appears to be called after and only for display tasks. So I thought it best to be careful.

Reply:
setting it `unattainable` early might cause an issue when we eventually run `markEnd` since the update will not continue down the tree past the task already marked unattainable. For example:
a <- b <- c
`c` depends on `b` which depends on `a`. If `b` is marked unattainable before we run markEnd on `a`, `c` will never be marked unattainable.

Reply:
Ah okay I see your point. I can just not update if the dependency exists then!